src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisonersearch/PrisonerSearchRequest.kt

```diff
@@ -0,0 +1,41 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.prisonersearch
+
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.barcode.BarcodeRecipient
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.stream.Collectors
+
+data class PrisonerSearchRequest(
+  val prisonNumber: String? = null,
+  val firstName: String?,
+  val lastName: String,
+  val dob: LocalDate? = null
+) {
+  constructor(barcodeRecipient: BarcodeRecipient) : this(
+    barcodeRecipient.prisonNumber,
+    barcodeRecipient.firstName,
+    barcodeRecipient.lastName,
+    barcodeRecipient.dob
+  )
+
+  fun toRequestBody(): Map<String, String?> =
+    mapOf(
+      "nomsNumber" to prisonNumber,
+      "firstName" to firstName,
+      "lastName" to lastName,
+      "dateOfBirth" to dob?.format(DateTimeFormatter.ISO_LOCAL_DATE)
+    )
+}
+
+private val BarcodeRecipient.firstName: String?
+  get() = this.name.split(" ").firstElementIfMoreThan1ElementElseNull()
+
+private val BarcodeRecipient.lastName: String
+  get() = this.name.split(" ").secondAndBeyondElementsIfMoreThan1ElementElseAllElements()
+
+private fun List<String>.firstElementIfMoreThan1ElementElseNull(): String? =
+  if (this.size > 1) this[0] else null
+
+private fun List<String>.secondAndBeyondElementsIfMoreThan1ElementElseAllElements(): String =
+  if (this.size > 1) this.stream().skip(1).collect(Collectors.joining(" ")) else this.stream()
+    .collect(Collectors.joining(" "))
```

Comment:
I'm not sure about these method names. They're easier to understand than code but still quite hard to comprehend...

Reply:
I wonder if it would be easier to grok if you had a method that returned a `Pair<String, String>`?

Reply:
OK, I can look at that in my next PR